test/Sentry.Tests/Internals/FileSystemTests.cs

```diff
@@ -0,0 +1,56 @@
+using System.IO.Abstractions.TestingHelpers;
+
+namespace Sentry.Tests.Internals;
+
+public class FileSystemTests
+{
+    private const string Filename = "lock.txt";
+
+    [Fact]
+    public void Leased_file_cannot_be_leased_again()
+    {
+        const string path = @$"C:\test\{Filename}";
+
+        IDictionary<string, MockFileData> files  = new Dictionary<string, MockFileData>
+        {
+            { path, new MockFileData(Filename) { AllowedFileShare = FileShare.None } }
+        };
+
+        IFileSystem fileSystem = new FakeFileSystem(files);
+
+        Assert.Throws<IOException>(() =>
+        {
+            using (fileSystem.GetLeaseFile(path)) { }
+        });
+    }
+
+    [Fact]
+    public void RealFileSystem_Leased_file_is_accessible_after_process_crush()
+    {
+        FileSystem.Instance.DeleteFile(Filename);
+
+        var thread = new Thread(() =>
+        {
+            _ = FileSystem.Instance.GetLeaseFile(Filename);
+        });
+
+        thread.Start();
+        thread.Join(200);
```

Comment:
Could this just be?

```suggestion
Task.Run(() => FileSystem.Instance.GetLeaseFile(Filename)).Wait();
```

Reply:
In theory, yes. I wanted to ensure it's a different thread. IIRC, tasks don't have to be on a different thread. They could be on the same thread, which would defeat the purpose of the validation.